src/app/services/baw-api/projects.service.ts

```diff
@@ -9,71 +9,49 @@ import {
 } from "src/app/interfaces/apiInterfaces";
 import { Project, ProjectInterface } from "src/app/models/Project";
 import { AppConfigService } from "../app-config/app-config.service";
-import { APIErrorDetails } from "./api.interceptor";
-import { BawApiService, Filters } from "./base-api.service";
+import { Filters } from "./base-api.service";
+import { ModelService } from "./model.service";
 
 /**
  * Interacts with projects route in baw api
  */
 @Injectable({
   providedIn: "root"
 })
-export class ProjectsService extends BawApiService {
+export class ProjectsService extends ModelService<Project> {
+  private paths: {
+    [key: string]: string;
+  };
+
   constructor(http: HttpClient, config: AppConfigService) {
-    super(http, config);
+    const classBuilder = (project: ProjectInterface) => new Project(project);
```

Comment:
is this needed for every model?

Let's make it optional. Doing a `new T()` should be possible inside of `ModelService`

Reply:
Seems to be possible, although not as clean as you are thinking: https://stackoverflow.com/questions/17382143/create-a-new-object-from-type-parameter-in-generic-class

Reply:
Yep, you're right.

I'd suggest following this pattern:

```typescript
//model
class foo {
    constructor() {
    }

    name: string;
}

abstract class CommonApi<T> {
    abstract modelFactory(): T;
    

    get() {
        return this.modelFactory();
    }
}

class FooService extends CommonApi<foo> {
    modelFactory() { return new foo() };

}
```

Reply:
I found what I consider to be a cleaner solution in this commit: https://github.com/QutEcoacoustics/workbench-client/pull/67/commits/fe59ca29ed64861a5096f2aa1226580fbf0a1341.

Basically it allows me to extend the `ApiCommon` class like this:

```javascript
class ApiCommon<T> {
  constructor(
    http: HttpClient,
    config: AppConfigService,
    private type: new (object: any) => T
  ) {
    super(http, config);
  }
}

class ProjectsService extends ApiCommon<Project> {
  constructor(http: HttpClient, config: AppConfigService) {
    super(http, config, Project);
  }
}
```